bokeh/charts/chart.py

```diff
@@ -82,14 +82,14 @@ class Chart(Plot):
     def __init__(self):
 
         # Initializes then gets default properties
-        super(Chart, self).__init__(id=self._options.id or make_id())
+        # NOTE: we use option._id here since ChartOptions converts id arg to _id
+        super(Chart, self).__init__(id=self._options._id or make_id())
 
         # create new chart options to get default properties with values
         default_props = ChartOptions().properties_with_values()
 
         # get the properties and values from the module-wide defaults object
         option_props = ChartOptions.properties_with_values(defaults)
```

Comment:
> `defaults` is a default-constructed ChartOptions. So default_props and option_props should always be identical ? I guess the idea is that someone can change the defaults object?

Yes, should be identical. More on this below..

(as a side note I don't like the `defaults` alias for `default_options`. The later is much more explicit...)

> Why ChartOptions.properties_with_values(defaults) instead of defaults.properties_with_values()?

+1

> if bokeh.charts.chart_options.default_options has been modified, it looks like the values in there override options that are explicitly passed in to this Chart.__init__ - is that intentional? Shouldn't my constructor kwargs override the default_options?

I don't think so. 

In general I'm +1 on your suggestions and would implement these as part of this PR. I think I like the later proposal (regarding theme).

@rothnic @bryevdv , opinions?